### PR TITLE
Remove unused sources

### LINKED ISSUE #6546: Deprecate unused authentication sources
**Is your feature request related to a problem? Please describe.**
After discussion in our internal meeting, we should deprecate sources that aren't being used actively.

Current list of sources:

- Authorize.NET
- Mirapay
- Pinterest
- Twitter
- Instagram

**Describe the solution you'd like**

For each of the sources, we must:

- Remove the associated test in testlink
- Remove the backend auth source code
- Remove the formhandler forms
- Remove the code in the admin interface (vue)
- Create an upgrade script to remove any sources matching these types
- Add an upgrade note in our guide that states we did deprecate these

### PR DESCRIPTION
**Pull request by jrouzierinverse**
# Description
Remove the following sources AuthorizeNet, Instagram, Twitter, Pinterest, and  Mirapay.

# Impacts
Authorization Sources

# Issue
Fixes #6546

# Delete branch after merge
YES

# Checklist
- [ ] Document the feature
- [ ] Add unit tests

# NEWS file entries

## Enhancements
* Remove unused/deprecated sources

### CODE REVIEW

File: addons/upgrade/to-11.1-remove-unused-sources.pl [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+use pf::IniFiles;
+use pf::file_paths qw($authentication_config_file);
+use pf::util;
+
+my $file = $authentication_config_file;
+if (@ARGV) {
+    $file = $ARGV[0];
+}
+
+our %typesToDelete = map { $_ => undef  } qw(Twitter Pinterest Mirapay Instagram AuthorizeNet);
```
Comment (julsemaan) [inline addons/upgrade/to-11.1-remove-unused-sources.pl]: portal modules also need to be handled by this script

File: addons/upgrade/to-11.1-remove-unused-sources.pl [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+my $cs = pf::IniFiles->new(-file => $file, -allowempty => 1);
+my $update = 0;
+
+for my $section ( grep {/^\S+$/} $cs->Sections() ) {
+    my $type = $cs->val($section, 'type');
+    if (exists $typesToDelete{$type}) {
+        print "Removing $section\n";
+        $cs->DeleteSection($section);
+        for my $group ($cs->GroupMembers($section)) {
+            $cs->DeleteSection($group);
```
Comment (julsemaan) [inline addons/upgrade/to-11.1-remove-unused-sources.pl:80]: I don't think this script is handling the removal of the sources in the other configuration sections (like connection profiles)

**Comment by jrouzierinverse** [comment]
Done

**Review by satkunas — APPROVED** [review]
UI changes look good.

**Comment by julsemaan** [comment]
The code looks good, I've setup a pipeline to run the full test suite:
https://gitlab.com/inverse-inc/packetfence/-/pipelines/374567710

**Comment by julsemaan** [comment]
Looks like the tests failed in the pipeline, @jrouzierinverse, can you look at this?

**Comment by jrouzierinverse** [comment]
Looks like it is unrelated to my changes I will rebase to see that will fix the issue.

**Comment by jrouzierinverse** [comment]
rebased please try again.

**Comment by julsemaan** [comment]
> rebased please try again.

https://gitlab.com/inverse-inc/packetfence/-/pipelines/374671413

**Comment by extrafu** [comment]
@julsemaan Status on this one?

**Comment by julsemaan** [comment]
> @julsemaan Status on this one?

Based on our conversation during our last internal meeting, @jrouzierinverse and @nqb were supposed to consult together regarding the broken tests

**Comment by nqb** [comment]
I ran a new pipeline without Golang unit tests (Perl+Integration tests):

https://gitlab.com/inverse-inc/packetfence/-/pipelines/382041744

**Comment by nqb** [comment]
Integration tests succeed but unit tests related to i18n failed, see https://gitlab.com/inverse-inc/packetfence/-/jobs/1646875323#L702

I suppose this is related to remaining strings in *.po files.

**Comment by extrafu** [comment]
@jrouzierinverse Have you verified the failing test? We need to merge this ASAP.

**Comment by nqb** [comment]
Unit tests were failing because Perl tests were running on Debian in place of EL.
There is no issues, merging.

**Comment by nqb** [comment]
TestLink tests have been already deprecated.

**Pull request merged**